tests/test_connector_matrix.py

```diff
@@ -633,21 +637,21 @@ async def test_respond_image(self):
         ) as patched_upload:
 
             patched_upload.return_value = asyncio.Future()
-            patched_upload.return_value.set_result({"content_uri": "mxc://aurl"})
+            patched_upload.return_value.set_result([nio.UploadResponse({"content_uri": "mxc://aurl"})])
 
             patched_send.return_value = asyncio.Future()
             patched_send.return_value.set_result(None)
             await self.connector.send(image)
 
             patched_send.assert_called_once_with(
+                room_id="!test:localhost",
+                message_type="m.room.message",
                 content={
                     "body": "opsdroid_upload",
                     "info": {"w": 1, "h": 1, "mimetype": "image/gif", "size": 26},
                     "msgtype": "m.image",
-                    "url": "mxc://aurl",
+                    "url": {"content_uri": "mxc://aurl"},
```

Comment:
I don't think this change is correct. The form of the content dict here is just as described in the spec: https://matrix.org/docs/spec/client_server/r0.6.0#m-image unless I am missing something?